Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -300,4 +276,71 @@ ExceptionOr<std::optional<URLPatternResult>> URLPattern::exec(std::optional<URLP
     return Exception { ExceptionCode::NotSupportedError };
 }
 
+// https://urlpattern.spec.whatwg.org/#hostname-pattern-is-an-ipv6-address
+static bool isIPv6Address(StringView input)
+{
+    Vector<UChar32> inputCodepoints(input.length());
```

Comment:
There must be a way to write this function without having to copy input into a new buffer. This is very inefficient.

Also, there's already a likely more efficient `isIPv6Address()` in URL.cpp that we can likely reuse.

Reply:
Yes, there is. Had to refactor URL.cpp a little bit to use it though.